libs/barista-components/filter-field/src/filter-field-edit-mode.spec.ts

```diff
@@ -462,9 +473,33 @@ describe('DtFilterField', () => {
       expect(filterChangeEvent).toBeDefined();
       expect(filterChangeEvent!.added.length).toBe(0);
       expect(filterChangeEvent!.removed.length).toBe(1);
-      expect(filterChangeEvent!.filters.length).toBe(2);
+      expect(filterChangeEvent!.filters.length).toBe(3);
 
       sub.unsubscribe();
     });
+
+    it('should prefill the input on edit of a first layer free-text filter', () => {
+      const tags = fixture.debugElement.queryAll(
+        By.css('.dt-filter-field-tag-label'),
+      );
+      tags[1].nativeElement.click();
+      advanceFilterfieldCycle();
+
+      const inputField = getInput(fixture);
+
+      expect(inputField.value).toBe('Custom free text');
+    });
+
+    it('should not prefill the input on edit of a second layer free-text filter', () => {
+      const tags = fixture.debugElement.queryAll(
+        By.css('.dt-filter-field-tag-label'),
+      );
+      tags[3].nativeElement.click();
+      advanceFilterfieldCycle();
+
+      const inputField = getInput(fixture);
+
+      expect(inputField.value).not.toBe('Custom free text');
```

Comment:
Can we have a bit clearer expectation here. There are a lot of things in the world that are `not` _Custom free text_.

What do we expect here?